### ISSUE TITLE
Plant state unknown after update

### ISSUE THREAD

**Issue by MD0005** [Olen/homeassistant-plant#210]
Hello. After update to v2024.11 all plants based on esp32 sensors have changed the state to unknown and attributes to null. Downgrading didnt resolve the issue.
Strangely plants based on Mija sensor are working properly.

Im not sure how to recover the proper plant entities to more than 15 plants :(

![IMG-20241205-WA0017](https://github.com/user-attachments/assets/101b468f-c13b-441f-aa4e-cbe8f97446ca)
![IMG-20241205-WA0019](https://github.com/user-attachments/assets/c57dfc35-5c75-474a-8048-1df900a3155b)

**Comment by MD0005** [comment]
Actual sensor is fine. Showing value. The value is also shown by the Plant Monitor Soil Moisture 'virtual' sensor. The issue is with the plant.x entity that is not evaluating related sensors.

![IMG-20241205-WA0033](https://github.com/user-attachments/assets/b1fa79bf-924d-41fe-be3d-24dc36505575)
![IMG-20241205-WA0032](https://github.com/user-attachments/assets/292e884c-f7da-43aa-9505-5a6e265522da)

**Comment by WendelFelius** [comment]
Same here! 👎🏻

**Comment by Tamago2000** [comment]
Same here.

**Comment by Balooforever** [comment]
Same here...

**Comment by Olen** [comment]
It would be really nice if anyone could provide _slighty_ more info than "me too".
Some logs, some info about the sensor values, some debug output, your settings for what shold trigger a problem state...

**Comment by Balooforever** [comment]
Hi Olen,

My log : 
`Update for plant.aechmea fails
Traceback (most recent call last):
  File "/usr/local/lib/python3.13/site-packages/homeassistant/helpers/entity.py", line 960, in async_update_ha_state
    await self.async_device_update()
  File "/usr/local/lib/python3.13/site-packages/homeassistant/helpers/entity.py", line 1320, in async_device_update
    await hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/plant/__init__.py", line 745, in update
    and self.dli.state is not None
        ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/homeassistant/components/sensor/__init__.py", line 556, in state
    unit_of_measurement = self.unit_of_measurement
                          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.13/site-packages/homeassistant/components/sensor/__init__.py", line 537, in unit_of_measurement
    := self.platform.default_language_platform_translations.get(translation_key)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'default_language_platform_translations'`

**Comment by MD0005** [comment]
Underneath is my log. But I think I know what might be the cause. Im using ESP32 to get the conductivity and temperature. I have disabled all other entities under these plants that I was sure I will be not using (f.e. air humidity). I didnt wanted to keep so many artifical entities with status 'unknown'. With previous version it was working, with the new version, the main plant.x entity is not working with even one entity disabled.

If you have the same issue, after you enable back the other entities, remember to restart HA.



`Logger: homeassistant.helpers.entity
Source: helpers/entity.py:960
First occurred: 21:32:49 (1755 occurrences)
Last logged: 22:21:20

Update for plant.epipremnum_2 fails
Update for plant.kalanch fails
Update for plant.epipremnum_3 fails
Update for plant.epipremnum_4_bottom fails
Update for plant.euphorbia fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 960, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1320, in async_device_update
    await hass.async_add_executor_job(self.update)
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/plant/__init__.py", line 745, in update
    and self.dli.state is not None
        ^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 556, in state
    unit_of_measurement = self.unit_of_measurement
                          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 537, in unit_of_measurement
    := self.platform.default_language_platform_translations.get(translation_key)
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'default_language_platform_translations'`

**Comment by Olen** [comment]
Sounds like this can be related to the translateable units of measurements that I think are new.

Not sure how to fix that in a hurry

**Comment by eUwSqBTm** [comment]
I got mine working again by enabling all the disabled entities under sensors (my sensors only do temp/moisture) and then reinstalling the home assistant plant integration. 

My ESP32 ones that have all the sensors never stopped working.

**Comment by Olen** [comment]
Have you also updated the list of sensors that should trigger a "problem" status?
Could it be a mismatch between the sensors there and the sensors provided by the esp32 integration?

**Comment by klaptafel** [comment]
I only use the soil moisture as problem trigger, and only the Dli entities were disabled (all 3). This situation caused the 'unknown' state. Enabling the 3 Dli entities and restarting Home Assistant fixed it. 

However, ideally when not using the other categories as problem triggers it should be no issue to disable them (either manually or maybe even automatically after (re)configuring).

**Comment by WhimsySpoon** [comment]
I have 20+ devices using a MiFlora BLE sensor with no issue. Today I tried adding two plants with a [GIEX GX04](https://www.zigbee2mqtt.io/devices/GX04.html#giex-gx04). Unlike the MiFlora, the GIEX device only exposes Soil Moisture and Temperature.

I set up a plant with these two sensors only, and disabled the rest. This generated the errors others have reported above on HA Core start-up.

Following @klaptafel's advice, I started tinkering with the DLI entities and have discovered that only the sensor `_dli` is needed (not the two configuration ones. Following Olen's advice, I have also disabled all but the moisture and temperature sensors in the trigger options in the device's configuration.

In summary, the error appears to be that if this sensor entity is disabled, the device cannot be updated/started.

I hope this helps track down the issue,

**Comment by RBarth-DE** [comment]
> I have 20+ devices using a MiFlora BLE sensor with no issue. Today I tried adding two plants with a [GIEX GX04](https://www.zigbee2mqtt.io/devices/GX04.html#giex-gx04). Unlike the MiFlora, the GIEX device only exposes Soil Moisture and Temperature.
> 
> I set up a plant with these two sensors only, and disabled the rest. This generated the errors others have reported above on HA Core start-up.
> 
> Following [@klaptafel](https://github.com/klaptafel)'s advice, I started tinkering with the DLI entities and have discovered that only the sensor `_dli` is needed (not the two configuration ones. Following Olen's advice, I have also disabled all but the moisture and temperature sensors in the trigger options in the device's configuration.
> 
> In summary, the error appears to be that if this sensor entity is disabled, the device cannot be updated/started.
> 
> I hope this helps track down the issue,

Fixed it for me. Enabling Dli solved the update issue.
